src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -6569,250 +6571,85 @@ export class StudyViewPageStore
+        [id: string]: ChartMeta;
+    } {
+        let chartMetaSet = getChartMetaSetForClinicalAttributes(
+            this._customCharts,
+            this.molecularProfiles.result,
+            this._geneSpecificCharts,
+            this._genericAssayCharts,
+            this._XvsYCharts,
+            this.clinicalAttributes.result,
+            [],
+            this.mutationProfiles.result,
+            this.structuralVariantProfiles.result,
+            this.isStructVarTableFeatureEnabled,
+            this.cnaProfiles.result,
+            this.shouldDisplayClinicalEventTypeCounts.result,
+            this.shouldDisplaySampleTreatments.result,
+            this.shouldDisplayPatientTreatments.result,
+            this.shouldDisplaySampleTreatmentGroups.result,
+            this.shouldDisplayPatientTreatmentGroups.result,
+            this.shouldDisplaySampleTreatmentTarget.result,
+            this.shouldDisplayPatientTreatmentTarget.result
+        );
+        return chartMetaSet;
+    }
+
+    // all chart meta information
     @computed
     get chartMetaSet(): { [id: string]: ChartMeta } {
-        // Only add Mobx Promises that are not dependent on StudyViewFilter will force re-render
-        let _chartMetaSet = _.fromPairs(this._customCharts.toJSON());
-        if (_.isEmpty(this.molecularProfiles.result)) {
-            delete _chartMetaSet[
-                SpecialChartsUniqueKeyEnum.GENOMIC_PROFILES_SAMPLE_COUNT
-            ];
-        }
-        _chartMetaSet = _.merge(
-            _chartMetaSet,
-            _.fromPairs(this._geneSpecificCharts.toJSON()),
-            _.fromPairs(this._genericAssayCharts.toJSON()),
-            _.fromPairs(this._XvsYCharts.toJSON())
-        );
-
-        // Add meta information for each of the clinical attribute
-        // Convert to a Set for easy access and to update attribute meta information(would be useful while adding new features)
-        _.reduce(
-            this.chartClinicalAttributes.result,
-            (acc: { [id: string]: ChartMeta }, attribute) => {
-                const uniqueKey = getUniqueKey(attribute);
-                const priority = getPriorityByClinicalAttribute(attribute);
-                if (priority > -1) {
-                    acc[uniqueKey] = {
-                        displayName: attribute.displayName,
-                        uniqueKey: uniqueKey,
-                        dataType: getChartMetaDataType(uniqueKey),
-                        patientAttribute: attribute.patientAttribute,
-                        description: attribute.description,
-                        priority: priority,
-                        renderWhenDataChange: false,
-                        clinicalAttribute: attribute,
-                    };
-                }
-                return acc;
-            },
-            _chartMetaSet
-        );
-
-        _.reduce(
+        let chartMetaSet = getChartMetaSetForClinicalAttributes(
```

Comment:
lets put a comment explaining why the general (chartMetaSet) starts with the clinical attributes set.

Reply:
the general chartMetaSet doesn't start with the clinical attributes set; it just contains the meta information for clinical attributes. the function name is misleading, changed to `getChartMetaSet(...)`.